recipes/_update_shcluster.rb

```diff
@@ -0,0 +1,31 @@
+# coding: UTF-8
+#
+# Cookbook Name:: cerner_splunk
+# Recipe:: _update_shcluster
+#
+# Recipe determines whether to add/remove this SH member to/from SH Cluster.
+
+password_file = File.join node['splunk']['external_config_directory'], 'password'
+search_heads = CernerSplunk.my_cluster_data(node)['sh_cluster']
+
+fail 'Search Heads are not configured for sh clustering in the cluster databag' if search_heads.nil? || search_heads.empty?
+
+unless node['splunk']['config']['sh_member_action'].nil?
```

Comment:
This strongly feels like it should be a resource, possibly along with the initialize in the captain recipe... 

action :initialize "Check if I'm a member of a search head cluster, and create with all peers if I am not"
action :add "Check if I am a member of a search head cluster, and add me with one of the peers if I am not"
action :remove "Check if I am a member of a search head cluster, and remove me if I am"